### PR TITLE
Move the interface based on a canvas to the config Docker image

### PR DESCRIPTION
**Pull request by sbrunner**
The goal of this pull request is to be able to have the template of the desktop application base on the canvas (from desktop-alt) in the config container (for simple application).
The js and the css will come from ngeo directly.

### CODE REVIEW

File: geoportal/c2cgeoportal_geoportal/__init__.py [hunk 1]
```diff
@@ -117,28 +136,41 @@ def add_interface_ngeo(
...
+        config.add_view(
+            Entry,
+            attr="get_ngeo_index_vars",
+            route_name=f"{route_name}theme",
+            renderer=renderer,
+            permission=permission,
+        )
+    else:
+        assert interface_config is not None
+        entry = SimpleEntry(interface_config)
```
Comment (arnaud-morvan) [inline geoportal/c2cgeoportal_geoportal/__init__.py:166]: View should not been instanciated at config time but at request time.

File: Dockerfile [hunk 2]
```diff
@@ -185,11 +187,19 @@ ARG VERSION
 ENV VERSION=$VERSION
 
 COPY bin/npm-packages /usr/bin/
+
+WORKDIR /opt/c2cgeoportal/geoportal
+COPY geoportal/package.json ./
+
+# hadolint ignore=SC2046,DL3016
+RUN npm --no-optional --unsafe-perm --no-package-lock install && \
```
[the remark this answers is not in the document]
Comment (sbrunner) [reply Dockerfile:195]: I can move it in the base image, but it's not installed like that in the tool image (when we will remove the advance mode we can do a big cleanup :-) )

**Comment by sbrunner** [comment]
@arnaud-morvan this is ready to be reviewed again :-)

File: bin/import-ngeo-apps [hunk 3]
```diff
@@ -46,112 +45,150 @@ def _sub(pattern, repl, string, count=0, flags=0, required=True):
...
-    Import the ngeo apps files
-    """
+def main() -> None:
+    """Import the ngeo apps files."""
 
     parser = ArgumentParser(description="Import ngeo apps files")
 
     parser.add_argument("--html", action="store_true", help="Import the html template")
     parser.add_argument("--js", action="store_true", help="Import the javascript controller")
+    # Replace the dynamicUrl and interface meta by dynamically generate variable.
```
Comment (arnaud-morvan) [inline bin/import-ngeo-apps:55]:
```suggestion
    # Replace the dynamicUrl and interface meta by dynamically generated variable.
```

File: doc/integrator/extend_application.rst [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+have an attribute slot. The slot says where the component should be added::
+
+- ``header`` -> in the header part of the page.
+- ``data`` -> in the data panel on the left of the map.
+- ``tool-button`` -> in the tools on the right of the map.
+- ``tool-button-separate`` -> in the tools on the right of the map, for the shared button.
+- ``tool-<panel-name>`` -> in the tools panel on the right of the map, when the tool is activated.
+- ``footer-<panel-name>`` -> in the footer part of the page, when the panel is activated.
+
+In the ``vars.yaml`` file your interface should be declared like that::
```
Comment (arnaud-morvan) [inline doc/integrator/extend_application.rst:35]:
```suggestion
In the ``vars.yaml`` file your interface should be declared like that:
```

File: geoportal/c2cgeoportal_geoportal/__init__.py [hunk 5]
```diff
@@ -141,6 +172,40 @@ def add_interface_ngeo(
...
+    renderer = f"/etc/geomapfish/interfaces/{route_name}.html.mako"
+    config.add_route(route_name, route, request_method="GET")
+    # Permalink theme: recover the theme for generating custom viewer.js URL
+    config.add_route(
+        f"{route_name}theme",
+        f"{route}{'' if route[-1] == '/' else '/'}theme/{{themes}}",
+        request_method="GET",
+    )
+    view = partial(canvas_view, interface_config=interface_config)
+    # view.__name__ = "my_partial_view"
```
Comment (arnaud-morvan) [inline geoportal/c2cgeoportal_geoportal/__init__.py:193]: remaining code comment

**Review by arnaud-morvan — APPROVED** [review]

**Pull request merged**